taskiq/cli/scheduler/run.py

```diff
@@ -142,32 +149,53 @@ async def run_scheduler_loop(scheduler: TaskiqScheduler) -> None:
     """
     loop = asyncio.get_event_loop()
     running_schedules = set()
+
+    interval_tasks_last_run: "Dict[str, datetime]" = {}
```

Comment:
Since it's a `dict[str, datetime]` it seems like if you are going to have multiple intervals, you solution will most probably crumble apart. You need to have a nested list (because intervals can have the same value). Each list item will be an interval value and last run.